### DISCUSSION TITLE
Sign-in flow is broken in GitHub Codespaces?

### DISCUSSION THREAD

**Discussion by haltcase** [nextauthjs/next-auth#7783] [Help]
It seems that the sign-in flow breaks under GitHub Codespaces, but I'm not sure if this is something this project can fix or if Codespaces' behavior would need to change. What I believe is happening is:

1. `getCsrfToken` is attempting to retrieve the `{ csrfToken: string }` payload from `/api/auth/csrf`
2. GitHub Codespaces has an intermediate HTML redirect page (due to its port forwarding) which is being returned instead of the expected JSON payload
3. Upon trying to parse this HTML as JSON, a `SyntaxError` is thrown and the sign-in fails, redirecting me back to the sign-in page

<details>
<summary>syntax error log</summary>

```
[next-auth][error][CLIENT_FETCH_ERROR] 
https://next-auth.js.org/errors#client_fetch_error Unexpected token '<', "<!doctype "... is not valid JSON {
  error: {
    message: `Unexpected token '<', "<!doctype "... is not valid JSON`,
    stack: `SyntaxError: Unexpected token '<', "<!doctype "... is not valid JSON\n` +
      '    at JSON.parse (<anonymous>)\n' +
      '    at parseJSONFromBytes (node:internal/deps/undici/undici:6613:19)\n' +
      '    at successSteps (node:internal/deps/undici/undici:6587:27)\n' +
      '    at node:internal/deps/undici/undici:1236:60\n' +
      '    at node:internal/process/task_queues:140:7\n' +
      '    at AsyncResource.runInAsyncScope (node:async_hooks:206:9)\n' +
      '    at AsyncResource.runMicrotask (node:internal/process/task_queues:137:8)\n' +
      '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)',
    name: 'SyntaxError'
  },
  url: 'https://haltcase-random-codespace-name-58x37oekt8441lub-3000.preview.app.github.dev/api/auth/csrf',
  message: `Unexpected token '<', "<!doctype "... is not valid JSON`
}
```
</details>

I've got `NEXTAUTH_URL` set as follows to avoid `ERR_INVALID_URL`:

```env
NEXTAUTH_URL=https://${CODESPACE_NAME}-3000.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}
```

Does anyone develop using `next-auth` inside GitHub Codespaces and has encountered this error?

**Comment by RemeJuan** [comment]
Hi there,

Did you ever figure this one out, just started tinkering with spaces yesterday and am running into the same issue

**Reply by haltcase** [reply]
Nope. It's been a while so I don't remember fully what I did to get around it, but I think it involved not using middleware at all and using checking auth per route instead.
